qutebrowser/browser/inspector.py

```diff
@@ -52,18 +54,50 @@ class WebInspectorError(Exception):
 
 class AbstractWebInspector(QWidget):
 
-    """A customized WebInspector which stores its geometry."""
+    """A customized WebInspector which stores its geometry.
 
-    def __init__(self, parent=None):
+    Attributes:
+        _position: position of the inspector (right/left/top/bottom/window)
+        _splitter: InspectorSplitter where the inspector can be placed.
+
+    Signals:
+        closed: Emitted when the inspector is closed.
+    """
+
+    closed = pyqtSignal()
+    _position = None
+    _splitter = None
+
+    def __init__(self, splitter, parent=None):
         super().__init__(parent)
         self._widget = None
         self._layout = miscwidgets.WrapperLayout(self)
-        self._load_state_geometry()
+        self._splitter = splitter
 
     def _set_widget(self, widget):
         self._widget = widget
         self._layout.wrap(self, widget)
 
+    def set_position(self, position):
+        """Set the position of the inspector."""
```

Comment:
Suggest adding "will close the inspector if position is None" and type hints.

Reply:
I added the comment.
I'm not up to date with modern python. How would I do the type hints?

Reply:
`position: typing.Optional[str]) -> None:`. For attr.ib or variables `# type: abc` is used to maintain compatibility with older Python version.